tests/reftests/admin-cache.unix.test

```diff
@@ -0,0 +1,406 @@
+  grep -q synopsis $op || echo 'synopsis: "A word"' >> $op
+  grep -q description $op || echo 'description: "Two words."' >> $op
+  grep -q authors $op || echo 'authors: "the testing team"' >> $op
+  grep -q homepage $op || echo 'homepage: "egapemoh"' >> $op
+  grep -q maintainer $op || echo 'maintainer: "maint@tain.er"' >> $op
+  grep -q license $op || echo 'license: "MIT"' >> $op
+  grep -q dev-repo $op || echo 'dev-repo: "hg+https://pkg@op.am"' >> $op
+  grep -q bug-reports $op || echo 'bug-reports: "https://nobug"' >> $op
+done
+### sh add-content.sh
+### : add archives
+### tar czf arch-ipsum.tgz add-content.sh
+### tar czf arch-dolor.tgz repo
+### tar czf arch-sit.tgz OPAM/config
+### tar czf arch-amet.tgz OPAM/repo/repos-config
+### <add-urls.sh>
+for n in $@; do
+  nv=$n.1
+  arch=arch-$n.tgz
+  file="packages/$n/$nv/opam"
+  echo "url {" >> $file
+  echo "src: \"$arch\"" >> $file
+  MD5=$(openssl md5 $arch | cut -d' ' -f2)
+  echo "checksum: \"md5=$MD5\"" >> $file
+  echo "}" >> $file
+done
+### sh add-urls.sh ipsum dolor sit amet
+### opam admin add-hashes sha256
+### opam admin add-hashes sha512
+[file://${BASEDIR}/arch-amet.tgz] found in cache
+[file://${BASEDIR}/arch-dolor.tgz] found in cache
+[file://${BASEDIR}/arch-ipsum.tgz] found in cache
+[file://${BASEDIR}/arch-sit.tgz] found in cache
+### rm -rf cache
+### : computes hashes
+### openssl md5 arch-ipsum.tgz | '.*= ' -> '' >$ IP_MD5
+### sh -c "echo '$IP_MD5' | cut -c 1-2" >$ PRE_IP_MD5
+### openssl sha256 arch-ipsum.tgz | '.*= ' -> '' >$ IP_SHA256
+### sh -c "echo '$IP_SHA256' | cut -c 1-2" >$ PRE_IP_SHA256
+### openssl sha512 arch-ipsum.tgz | '.*= ' -> '' >$ IP_SHA512
+### sh -c "echo '$IP_SHA512' | cut -c 1-2" >$ PRE_IP_SHA512
+### openssl md5 arch-dolor.tgz | '.*= ' -> '' >$ DL_MD5
+### sh -c "echo '$DL_MD5' | cut -c 1-2" >$ PRE_DL_MD5
+### openssl sha256 arch-dolor.tgz | '.*= ' -> '' >$ DL_SHA256
+### sh -c "echo '$DL_SHA256' | cut -c 1-2" >$ PRE_DL_SHA256
+### openssl sha512 arch-dolor.tgz | '.*= ' -> '' >$ DL_SHA512
+### sh -c "echo '$DL_SHA512' | cut -c 1-2" >$ PRE_DL_SHA512
+### openssl md5 arch-sit.tgz | '.*= ' -> '' >$ ST_MD5
+### sh -c "echo '$ST_MD5' | cut -c 1-2" >$ PRE_ST_MD5
+### openssl sha256 arch-sit.tgz | '.*= ' -> '' >$ ST_SHA256
+### sh -c "echo '$ST_SHA256' | cut -c 1-2" >$ PRE_ST_SHA256
+### openssl sha512 arch-sit.tgz | '.*= ' -> '' >$ ST_SHA512
+### sh -c "echo '$ST_SHA512' | cut -c 1-2" >$ PRE_ST_SHA512
+### openssl md5 arch-amet.tgz | '.*= ' -> '' >$ MT_MD5
+### sh -c "echo '$MT_MD5' | cut -c 1-2" >$ PRE_MT_MD5
+### openssl sha256 arch-amet.tgz | '.*= ' -> '' >$ MT_SHA256
+### sh -c "echo '$MT_SHA256' | cut -c 1-2" >$ PRE_MT_SHA256
+### openssl sha512 arch-amet.tgz | '.*= ' -> '' >$ MT_SHA512
+### sh -c "echo '$MT_SHA512' | cut -c 1-2" >$ PRE_MT_SHA512
+### <analysi.sh>
+pkg=$1
+kind=$2
+hash=$3
+
+pre=`echo $hash | cut -c 1-2`
+arch=cache/$kind/$pre/$hash
+msg="$pkg : $kind :"
+
+if [ -L $arch ]; then
+  link=`readlink "$arch"`
```

Comment:
I'm not sure if readlink is portable enough. I need here to retrieve the target of the link, not the fully resolved one